Runner/JobManager.py

```diff
@@ -0,0 +1,332 @@
+    str(datetime.datetime.now().minute)
+
+
+async def submit_job(batch_service_client, template, parameters):
+    """
+    Submits a Job against the batch service.
+
+    :param batch_client: The batch client to use.
+    :type batch_client: `batchserviceclient.BatchServiceClient`
+    :param str template: the json desciption of the job
+    :param str the parameters of the job
+    """
+    try:
+        loop = asyncio.get_event_loop()
+        job_json = batch_service_client.job.expand_template(
+            template, parameters)
+        jobparameters = batch_service_client.job.jobparameter_from_json(
+            job_json)
+        job = await loop.run_in_executor(None, functools.partial(batch_service_client.job.add, jobparameters))
+    except batchmodels.batch_error.BatchErrorException as err:
+        Utils.logger.info(
+            "Failed to submit job\n{}\n with params\n{}".format(
+                template, parameters))
+        traceback.print_exc()
+        Utils.print_batch_exception(err)
+        raise
+
+
+class JobManager(object):
+
+    def __init__(self, template_file, pool_template_file,
+                 parameters_file, expected_output, application_licenses=None):
+        super(JobManager, self).__init__()
+        self.raw_job_id = ctm.get_job_id(parameters_file)
+        self.job_id = _time + "-" + self.raw_job_id
+        self.pool_id = ctm.get_pool_id(parameters_file)
+        self.template_file = template_file
+        self.parameters_file = parameters_file
+        self.application_licenses = application_licenses
+        self.expected_output = expected_output
+        self.pool_template_file = pool_template_file
+        self.storage_info = None
+        self.job_status = Utils.JobStatus(
+            Utils.JobState.NOT_STARTED,
+            "Job hasn't started yet.")
+        self.duration = None
+
+    def __str__(self) -> str:
+        return "job_id: [{}] pool_id: [{}] ".format(self.job_id, self.pool_id)
+
+    async def create_and_submit_Job(self, batch_service_client):
+        """
+        Creates the Job that will be submitted to the batch service
+
+        :param batch_client: The batch client to use.
+        :type batch_client: `batchserviceclient.BatchServiceClient`
+        """
+        Utils.logger.info('Creating Job [{}]... job will run on [{}]'.format(self.job_id, self.pool_id))
+
+        # load the template and parameters file
+        template = ctm.load_file(self.template_file)
+        parameters = ctm.load_file(self.parameters_file)
+
+        # overrides some of the parameters needed in the file, container SAS
+        # tokens need to be generated for the container
+        ctm.set_parameter_name(parameters, self.job_id)
+        ctm.set_parameter_storage_info(parameters, self.storage_info)
+
+        # Submits the job
+        await submit_job(batch_service_client, template, parameters)
+
+    async def create_pool(self, batch_service_client, image_references):
```

Comment:
💡  Recommnedation: You can always add async to the name of the async funciton like here this funct can be : `create_pool_async`.

Reply:
Yep i'll do this